File: employee-management-api/src/main/java/com/harish/Employee/service/EmployeeService.java
package com.harish.Employee.service;

import java.util.List;

import com.harish.Employee.model.Employee;

public interface EmployeeService {

    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployee();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployee(Long id, Employee employee);
    
}

File: employee-management-ui/src/services/EmployeeService.js
import axios from "axios";
import React from "react";

const EMPLOYEE_API_BASE_URL="http://localhost:8080/api/v1/employees"

class EmployeeService extends React.Component{

    saveEmployee(employee){
        return axios.post(EMPLOYEE_API_BASE_URL,employee);
    }

    getEmployees()
    {
        return axios.get(EMPLOYEE_API_BASE_URL);
    }

    deleteEmployee(id){
        return axios.delete(EMPLOYEE_API_BASE_URL+"/"+id);
    }

    getEmployeeById(id){
        return axios.get(EMPLOYEE_API_BASE_URL+"/"+id);
    }

    updateEmployee(employee,id){
        return axios.put(EMPLOYEE_API_BASE_URL+"/"+id,employee);
    }
}

// eslint-disable-next-line import/no-anonymous-default-export
export default new EmployeeService();